sklearn/metrics/_pairwise_distances_reduction/_argkmin.pyx.tp

```diff
@@ -98,8 +98,14 @@ cdef class ArgKmin{{name_suffix}}(BaseDistanceReducer{{name_suffix}}):
                 metric_kwargs=metric_kwargs,
             )
         else:
-             # Fall back on a generic implementation that handles most scipy
-             # metrics by computing the distances between 2 vectors at a time.
+            # Fall back on a generic implementation that handles all distance
+            # metrics by computing it between 2 vectors at a time.
+
+            # The extra `Y_norm_squared` argument for the back-end is only
+            # supported for the Eucliean variant.
+            if metric_kwargs is not None:
+                metric_kwargs.pop("Y_norm_squared", None)
```

Comment:
I'm unsure who should responsible for doing this. The options I see are:

1. The caller does only passes `Y_norm_squared` if the metric is euclidean.
2. `Y_norm_squared` is still passed in, but the metric can choose to ignore it.
3. This PR, where the dispatcher removes it. For me, it's a little strange to have the dispatcher know about `Y_norm_squared`.

Reply:
c37e61ed145fbbbbc1d54a0333c9e980156a70a7 implements option 2.